// @snyk/error-catalog-nodejs-public/src/catalogs/error-codes.js
/*
 * © 2023 Snyk Limited
 *
 * Licensed under the Apache License, Version 2.0 (the 'License');
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an 'AS IS' BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = {
    Snyk: {
        TooManyRequests: 'SNYK-0001',
        NotImplemented: 'SNYK-0002',
        BadRequest: 'SNYK-0003',
        Timeout: 'SNYK-0004',
        Unauthorised: 'SNYK-0005',
        Server: 'SNYK-9999',
    },
    OpenSourceEcosystems: {
        UnparseableManifest: 'SNYK-OS-0001',
        UnparseableLockFile: 'SNYK-OS-0002',
        UnknownDependencyVersion: 'SNYK-OS-0003',
        MissingHeader: 'SNYK-OS-0004',
        MissingPayload: 'SNYK-OS-0005',
        UnprocessableFile: 'SNYK-OS-0006',
        CannotGetFileFromSource: 'SNYK-OS-0007',
        MissingEnvironmentVariable: 'SNYK-OS-0008',
        UnsupportedManifestFile: 'SNYK-OS-DOTNET-0001',
        UnsupportedTargetFramework: 'SNYK-OS-DOTNET-0002',
        MissingStaticMainFunction: 'SNYK-OS-DOTNET-0003',
        PublishFailed: 'SNYK-OS-DOTNET-0004',
        PrivateModule: 'SNYK-OS-GO-0001',
        GoModFileMissing: 'SNYK-OS-GO-0002',
        SsoReAuthRequired: 'SNYK-OS-GO-0003',
        IncompleteProject: 'SNYK-OS-GO-0004',
        InconsistentVendoring: 'SNYK-OS-GO-0005',
        UnsupportedExternalFileGenerationSCM: 'SNYK-OS-GO-0006',
        MissingRequirementFromPom: 'SNYK-OS-MAVEN-0001',
        UnableToResolveValueForProperty: 'SNYK-OS-MAVEN-0002',
        UnableToResolveVersionForProperty: 'SNYK-OS-MAVEN-0003',
        CyclicPropertyDetectedInPomFile: 'SNYK-OS-MAVEN-0004',
        UnableToParseXML: 'SNYK-OS-MAVEN-0005',
        InvalidCoordinates: 'SNYK-OS-MAVEN-0006',
        SkippedGroup: 'SNYK-OS-MAVEN-0007',
        PomFileNotFound: 'SNYK-OS-MAVEN-0008',
        MissingProjectFromPom: 'SNYK-OS-MAVEN-0009',
        CannotResolveTargetPomFromXml: 'SNYK-OS-MAVEN-0010',
        CannotResolveTargetPomFromRepo: 'SNYK-OS-MAVEN-0011',
        CannotGetBuildFileFromRepo: 'SNYK-OS-MAVEN-0012',
        CannotCreateGitHost: 'SNYK-OS-MAVEN-0013',
        NoReleasedVersionForVersionsRange: 'SNYK-OS-MAVEN-0014',
        SourceNotSupported: 'SNYK-OS-MAVEN-0015',
        TimeoutWhenProcessingTheDepTree: 'SNYK-OS-MAVEN-0016',
        NoRepoFoundForTheNPMPackage: 'SNYK-OS-NODEJS-0001',
        CouldNotParseNPMRegistryURL: 'SNYK-OS-NODEJS-0002',
        CouldNotFindBrokerURL: 'SNYK-OS-NODEJS-0003',
        UnableToReplaceBrokerURL: 'SNYK-OS-NODEJS-0004',
        BadNPMVersion: 'SNYK-OS-NODEJS-0005',
        UnknownBlobEncodingOnGithub: 'SNYK-OS-NODEJS-0006',
        NoResultsFromForkerProcesses: 'SNYK-OS-NODEJS-0007',
        ChildProcessExecution: 'SNYK-OS-NODEJS-0008',
        NoValidPackageUpgrades: 'SNYK-OS-NODEJS-0009',
        NoDependencyUpdates: 'SNYK-OS-NODEJS-0010',
        CouldNotParseJSONFile: 'SNYK-OS-NODEJS-0011',
        Base64Encode: 'SNYK-OS-NODEJS-0012',
        Base64Decode: 'SNYK-OS-NODEJS-0013',
        MissingSupportedFile: 'SNYK-OS-NODEJS-0014',
        InvalidConfiguration: 'SNYK-OS-NODEJS-0015',
        UnsupportedRequirementsFile: 'SNYK-OS-PIP-0001',
        TooManyManifestFiles: 'SNYK-OS-PIP-0002',
        FailedToApplyDependencyUpdates: 'SNYK-OS-PIP-0003',
    },
    PurlVulnerabilityFetching: {
        OrganizationNotWhitelisted: 'SNYK-OSSI-1040',
        AuthorizationRequestFailure: 'SNYK-OSSI-1050',
        InvalidPurl: 'SNYK-OSSI-2010',
        NamespaceNotProvided: 'SNYK-OSSI-2011',
        UnsupportedEcosystem: 'SNYK-OSSI-2020',
        MissingComponent: 'SNYK-OSSI-2021',
        ComponentNotSupported: 'SNYK-OSSI-2022',
        UnsupportedGoVersionFormat: 'SNYK-OSSI-2023',
        PackageNotFound: 'SNYK-OSSI-2030',
        VulnerabilityServiceUnavailable: 'SNYK-OSSI-2031',
        VulnDBInvalidResponse: 'SNYK-OSSI-2032',
        VulndbNext: 'SNYK-OSSI-2033',
        InternalServer: 'SNYK-OSSI-2040',
        InvalidPaginationParameters: 'SNYK-OSSI-2041',
        TooManyPurls: 'SNYK-OSSI-2042',
        TooManyIssues: 'SNYK-OSSI-2043',
        UndefinedContainerDistro: 'SNYK-OSSI-2044',
        UnsupportedDebianDistro: 'SNYK-OSSI-2045',
        UndefinedContainerVendor: 'SNYK-OSSI-2046',
        UnsupportedContainerVendor: 'SNYK-OSSI-2047',
    },
    IsolatedBuilds: {
        InvalidRequest: 'SNYK-OS-8001',
        BuildEnvironmentNotFound: 'SNYK-OS-8002',
        UnsupportedEcosystem: 'SNYK-OS-8003',
    },
    OpenSourceProjectSnapshots: {
        InvalidRequest: 'SNYK-OSSI-OSPSS-1001',
        InvalidResponse: 'SNYK-OSSI-OSPSS-1002',
        DataTransformation: 'SNYK-OSSI-OSPSS-2001',
        StorageFailure: 'SNYK-OSSI-OSPSS-3001',
        InternalServer: 'SNYK-OSSI-OSPSS-4001',
    },
    OpenSourceProjectIssues: {
        InvalidRequest: 'SNYK-OSSI-OSPI-1001',
        InvalidResponse: 'SNYK-OSSI-OSPI-1002',
        DataTransformation: 'SNYK-OSSI-OSPI-2001',
        StorageFailure: 'SNYK-OSSI-OSPI-3001',
        InternalServer: 'SNYK-OSSI-OSPI-4001',
    },
    OpenAPI: {
        BadRequest: 'SNYK-OPENAPI-0001',
        Forbidden: 'SNYK-OPENAPI-0002',
        NotAcceptable: 'SNYK-OPENAPI-0003',
        NotFound: 'SNYK-OPENAPI-0004',
        MethodNotAllowed: 'SNYK-OPENAPI-0005',
        RequestEntityTooLarge: 'SNYK-OPENAPI-0006',
        Unauthorized: 'SNYK-OPENAPI-0007',
        UnsupportedMediaType: 'SNYK-OPENAPI-0008',
    },
    OpenSourceUnmanaged: {
        MavenSearchServiceUnavailable: 'SNYK-OSJVM-001',
        Sha1NotFound: 'SNYK-OSJVM-002',
    },
    SbomExport: {
        InternalServer: 'SNYK-OS-9000',
        UnexpectedDepGraphResponse: 'SNYK-OS-9001',
        UnexpectedParseDepGraph: 'SNYK-OS-9002',
        IaCOrSASTProject: 'SNYK-OS-9003',
        UnsupportedProject: 'SNYK-OS-9004',
        DepGraphResponse: 'SNYK-OS-9005',
        MissingAuthToken: 'SNYK-OS-9006',
        EmptyRequestBody: 'SNYK-OS-9007',
        InvalidDepGraph: 'SNYK-OS-9008',
    },
    Fix: {
        FailedToGetPullRequestAttributes: 'SNYK-PR-TEMPLATE-0001',
        PullRequestTemplateNotFound: 'SNYK-PR-TEMPLATE-0002',
        FailedToCompilePrTemplate: 'SNYK-PR-TEMPLATE-0003',
        FailedToParsePullRequestAttributes: 'SNYK-PR-TEMPLATE-0004',
        FailedToLoadCompiledYaml: 'SNYK-PR-TEMPLATE-0005',
        FailedToGenerateHash: 'SNYK-PR-TEMPLATE-0006',
        FailedToCreatePRTemplate: 'SNYK-PR-TEMPLATE-0007',
        FailedToReadPRTemplate: 'SNYK-PR-TEMPLATE-0008',
        FailedToDeletePRTemplate: 'SNYK-PR-TEMPLATE-0009',
        PRTemplateInvalidPayload: 'SNYK-PR-TEMPLATE-0010',
    },
};
